test.c

```diff
@@ -49,18 +43,46 @@ int main() {
   ok = hasUniqueChars(string3);
   assert(ok);
   
-  // TODO: add your tests here
+  // Test 3: Tests special characters. They are all in the ASCII range 32-127 and none are duplicates of each other, 
+  //so if any character fails the test, something is wrong.
+  strcpy(string3, "{}!@|~'&$#?><=;;/.,*()");
+  ok = hasUniqueChars(string3);
+  assert(ok);
+
+  //Test 4: Arbitrary string which should pass.
+  strcpy(string3, "a s hdfm,~");
+  ok = hasUniqueChars(string3);
+  assert(ok);
 
+  //Test 5: repeating character 'a'
+  strcpy(string3, "a !@#$a");
+  ok = hasUniqueChars(string3);
+  assert(!(ok));
 
+  //Test 6: testing more non-ASCII characters
+  strcpy(string3, "こんにちわ");
```

Comment:
Those look like unicode, not ascii, so they may be chopped off to partial symbols